File: quiz-backend/Quiz/PerguntasViewModel.cs
using System.Collections.Generic;

namespace Quiz
{
    public class PerguntasViewModel
    {
        public int Id { get; set; }
        public IEnumerable<PerguntaViewModel> Perguntas { get; set; } = new List<PerguntaViewModel>();
    }

    public class PerguntaViewModel
    {
        public int Id { get; set; }
        public string Enunciado { get; set; }
        public IEnumerable<RespostaViewModel> Respostas { get; set; } = new List<RespostaViewModel>();
    }

    public class RespostaViewModel
    {
        public int Id { get; set; }
        public string Texto { get; set; }
        public bool Selecionada { get; set; }
    }
}

File: quiz/src/app/perguntas/perguntas.viewmodel.ts
export class PerguntasViewModel {
    Id: number;
    Perguntas: PerguntaViewModel[];
}

export class PerguntaViewModel {
    Id: number;
    Enunciado: string;
    Respostas: RespostaViewModel[];
}

export class RespostaViewModel {
    Id: number;
    Texto: string;
    Selecionada: boolean = false;
}